core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -110,17 +110,28 @@ oppia.directive('searchBar', [function() {
         };
 
         var _searchBarFullyLoaded = false;
+        var url;
 
         var _hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
-        var _onSearchQueryChangeExec = function() {
-          searchService.executeSearchQuery(
+        var _onSearchQueryChangeExec = function(reloadPage) {
+          $scope.searchIsLoading = true;
+          url = searchService.executeSearchQuery(
```

Comment:
made a new function in `searchService` which returns a url. this variable now references that function instead.